tests/transaction_tests.cpp

```diff
@@ -100,22 +100,71 @@ TEST_CASE("Transaction guard") {
 
     storage.insert(Object{0, "Jack"});
 
-    SECTION("insert, call make a storage to call an exception and check that rollback was fired") {
-        auto countBefore = storage.count<Object>();
-        try {
-            auto guard = storage.transaction_guard();
+    struct ExceptionMatcher : Catch::Matchers::MatcherGenericBase {
+        std::string what;
```

Comment:
Checking the error code alone would be enough, we don't care about the exact message string at this point...

Reply:
fixed